test/base.h

```diff
@@ -54,27 +55,52 @@ void compute(std::string&& filename, std::vector<size_t> homology, std::string o
 	std::cout << "All good." << std::endl;
 }
 
-void run_all(bool full = false) {
-	compute("../../test/d2.flag", {{1ul, 1ul}});
-	compute("../../test/a.flag", {{1ul, 2ul, 0ul}});
-	compute("../../test/b.flag", {{1ul, 0ul, 0ul}});
-	compute("../../test/c.flag", {{1ul, 5ul}});
-	compute("../../test/d.flag", {{1ul, 0ul, 1ul}});
-	compute("../../test/e.flag", {{1ul, 0ul, 0ul, 0ul}});
-	compute("../../test/f.flag", {{1ul, 0ul, 0ul}});
-	compute("../../test/d3.flag", {{1ul, 0ul, 2ul}});
-	compute("../../test/d3-allzero.flag", {{1ul, 0ul, 2ul}});
-	compute("../../test/double-d3.flag", {{1, 0, 5}});
-	compute("../../test/double-d3-allzero.flag", {{1, 0, 5}});
-	compute("../../test/d4.flag", {{1, 0, 0, 9}});
-	compute("../../test/d4-allzero.flag", {{1, 0, 0, 9}});
-	compute("../../test/d5.flag", {{1, 0, 0, 0, 44}});
-	compute("../../test/d7.flag", {{1, 0, 0, 0, 0, 0, 1854}});
+void run_all(bool full = false, bool in_memory = false) {
```

Comment:
How about creating a run_all that depends on a type of flag complex computer just like you did for compute? Then you can call run_all<dfc_memory>(true/false) etc. If you want to hide that complexity you can still crate a wrapper that does the translation of bool into type, but probably not necessary.

Reply:
Nice idea, I didn't think about it but it makes thing much clear :)

Reply:
I pushed some commits, would be ready to merge. I hope I did not accidentally prevent some additions that you were still planning, let me know when you want to merge!

Reply:
Nothing more to add on my side.
When you're good on your side, I think that this PR can be merged :+1: